src/lib/autoinstall/clients/inst_autosetup_upgrade.rb

```diff
@@ -226,6 +226,23 @@ def main
           false
         )
 
+        #
+        # Checking if at least one base product has been selected/evaluated.
+        #
+        begin
+          Product.FindBaseProducts
+        rescue StandardError
+          msg = _("No new base product has been set.\n" \
+           "Please check the <b>software</b>/<b>products</b> entry in the " \
+           "AutoYaST configuration file.<br><br>" \
+           "Following base products are available:<br>")
+          Yast::AutoinstFunctions.available_base_products.each do |product|
+            msg += "#{product.details.product} (#{product.details.summary})<br>"
```

Comment:
Just be careful that `product.details` is available only for the Full installation medium!

This method can actually return an Y2Packager::Resolvable (openSUSE Leap/TW) or Y2Packager::ProductControlProduct (SLE Online medium) or  Y2Packager::ProductLocation (SLE Full medium) objects. Sorry, it's not that simple.... :worried: 

Please verify the functionality using all media types (SLE Online, SLE Full, openSUSE Leap 15.2).